streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -288,6 +291,8 @@ public StreamThread(TopologyBuilder builder,
         this.standbyRecords = new HashMap<>();
 
         this.stateDirectory = new StateDirectory(applicationId, config.getString(StreamsConfig.STATE_DIR_CONFIG), time);
+        Object maxPollInterval = consumerConfigs.get(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG);
+        this.rebalanceTimeoutMs =  maxPollInterval == null ? 300000 : (Integer) ConfigDef.parseType(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, maxPollInterval, Type.INT);
```

Comment:
Rather than this i think it would be better to add:
`tempConsumerDefaultOverrides.put(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, "300000");`
to the `CONSUMER_DEFAULT_OVERRIDES` map in `StreamsConfig`